Directory.Build.props

```diff
@@ -93,8 +93,8 @@
   <!-- Warnings and errors -->
   <PropertyGroup>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <!-- Don't make missing XML docs a fatal build error, but still surface so we have visibility into undocumented APIs. -->
-    <WarningsNotAsErrors>$(WarningsNotAsErrors);CS1591</WarningsNotAsErrors>
```

Comment:
Being unable to compile because a doc comment is missing particularly when we prototype can be very frustrating. Have we discussed this more broadly with the team since it has a dev inner loop impact?

Reply:
Can we just have this on for CI/official builds and off for local development maybe?  PR checks blocking merging seems like the best compromise to me

Reply:
That's a good point. I'm curious what we can do here.

I suppose we can make these errors only on the CI, but I think it could be annoying for devs to build locally but see the PR check fail. But I think it's probably more preferable than making these errors everywhere. Note that personally, I usually add empty doc comments for public API as I work and fill them in before opening a PR. But again, I know that's not how everyone likes to work.